nipype/utils/filemanip.py

```diff
@@ -449,9 +448,11 @@ def get_related_files(filename, include_this_file=True):
     path, name, this_type = split_filename(filename)
     for type_set in related_filetype_sets:
         if this_type in type_set:
-            for related_type in type_set:
-                if include_this_file or related_type != this_type:
-                    related_files.append(op.join(path, name + related_type))
+            related_files.extend(
+                op.join(path, name + related_type)
+                for related_type in type_set
+                if include_this_file or related_type != this_type
+            )
```

Comment:
This can be one big comprehension:

```python
path, name, this_type = split_filename(filename)
related_files = [
    op.join(path, f"{name}{related_type})
    for type_set in related_filetype_sets
    if this_type in type_set
    for related_type in type_set
    if include_this_file or related_type != this_type
]